includes/modules/pages/product_info/main_template_vars.php

```diff
@@ -1,158 +1,161 @@
+        $products_get_buy_now_qty = zen_get_buy_now_qty($_GET['products_id']);
 
-    $product_is_free = $product_info->fields['product_is_free'];
+        $products_base_price = $currencies->display_price(zen_get_products_base_price((int)$_GET['products_id']),
+          zen_get_tax_rate($product_info->fields['products_tax_class_id']));
 
-    $products_tax_class_id = $product_info->fields['products_tax_class_id'];
+        $product_is_free = $product_info->fields['product_is_free'];
 
-    $products_discount_type = $product_info->fields['products_discount_type'];
-    $products_discount_type_from = $product_info->fields['products_discount_type_from'];
-  }
+        $products_tax_class_id = $product_info->fields['products_tax_class_id'];
 
-  require(DIR_WS_MODULES . zen_get_module_directory('product_prev_next.php'));
+        $products_discount_type = $product_info->fields['products_discount_type'];
+        $products_discount_type_from = $product_info->fields['products_discount_type_from'];
 
-  $module_show_categories = PRODUCT_INFO_CATEGORIES;
-  $module_next_previous = PRODUCT_INFO_PREVIOUS_NEXT;
+        require(DIR_WS_MODULES . zen_get_module_directory('product_prev_next.php'));
 
-  $products_id_current = (int)$_GET['products_id'];
+        $module_show_categories = PRODUCT_INFO_CATEGORIES;
+        $module_next_previous = PRODUCT_INFO_PREVIOUS_NEXT;
 
-/**
- * Load product-type-specific main_template_vars
- */
-  $prod_type_specific_vars_info = DIR_WS_MODULES . 'pages/' . $current_page_base . '/main_template_vars_product_type.php';
-  if (file_exists($prod_type_specific_vars_info)) {
-    include_once($prod_type_specific_vars_info);
-  }
-  $zco_notifier->notify('NOTIFY_MAIN_TEMPLATE_VARS_PRODUCT_TYPE_VARS_PRODUCT_INFO');
+        $products_id_current = (int)$_GET['products_id'];
 
+        /**
+         * Load product-type-specific main_template_vars
+         */
+        $prod_type_specific_vars_info = DIR_WS_MODULES . 'pages/' . $current_page_base . '/main_template_vars_product_type.php';
+        if (file_exists($prod_type_specific_vars_info)) {
+            include_once($prod_type_specific_vars_info);
+        }
+        $zco_notifier->notify('NOTIFY_MAIN_TEMPLATE_VARS_PRODUCT_TYPE_VARS_PRODUCT_INFO');
+
+
+        /**
+         * Load all *.PHP files from the /includes/templates/MYTEMPLATE/PAGENAME/extra_main_template_vars
+         */
+        $extras_dir = $template->get_template_dir('.php', DIR_WS_TEMPLATE, $current_page_base . 'extra_main_template_vars', $current_page_base . '/' . 'extra_main_template_vars');
+        if ($dir = @dir($extras_dir)) {
+            while ($file = $dir->read()) {
+                if (!is_dir($extras_dir . '/' . $file)) {
+                    if (preg_match('~^[^\._].*\.php$~i', $file) > 0) {
+                        $directory_array[] = '/' . $file;
+                    }
+                }
+            }
+            $dir->close();
+        }
+        if (sizeof($directory_array)) {
```

Comment:
If the way "we" do it, this should use `count` instead of `sizeof`. Still not entirely sure why most/all are going this route other than `sizeof` is an alias to `count` and some previous "scare" that the alias was going to be removed.